client/explore/GenericExplorer.cs

```diff
@@ -58,7 +58,7 @@ public override ExplorerDecision<int> MapContext(PRG random, float[] weights, in
             {
                 weights[i] = weights[i] / total;
                 sum += weights[i];
-                if (sum > draw)
+                if (sum >= draw)

```

Comment:
why?

Reply:
If we don't do this, then (I believe) we won't be handling the maximum value correctly. Consider what happens when random.UniformUnitInterval() returns 1.0; in this case the sum of weights will never be greater than 1.0 (it will at most equal 1.0), so the for loop will exit naturally (no break). This will cause the correct actionIndex to be chosen, since we initialize actionIndex to the maximum value anyway, but the actionProbability will remain at 0, which leads to an incorrect probability. Also, even with respect to actionIndex it is brittle to make things work because of the initialized value and an unexpected exit from the loop -- if someone changes the loop code or the initial values, they will break this case. I can add comments to this effect.